test/common/compressor/zlib_compressor_impl_test.cc

```diff
@@ -47,6 +47,27 @@ TEST_F(ZlibCompressorImplDeathTest, CompressorTestDeath) {
   EXPECT_DEATH(unitializedCompressorTestHelper(), std::string{"assert failure: result == Z_OK"});
 }
 
+TEST_F(ZlibCompressorImplTest, CallingChecksum) {
+  const uint64_t expected_checksum{3587466910};
```

Comment:
I'm confused about where this magic number came from. My guess is that it relies on the random chars being deterministically generated because the RNG is explicitly seeded with certain values. I think it's OK, although I suppose the RNG implementation could change and break the test. A comment here about this would be a good idea.